atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -0,0 +1,816 @@
+        }
+        for (Iterator<Entry<Cell, OverflowValue>> entryIter = overflowResults.entrySet().iterator(); entryIter.hasNext(); ) {
+            Entry<Cell, OverflowValue> entry = entryIter.next();
+            Value value = results.get(entry.getKey());
+            if (value != null && value.getTimestamp() > entry.getValue().ts) {
+                entryIter.remove();
+            }
+        }
+        if (!overflowResults.isEmpty()) {
+            Map<Long, byte[]> values = Maps.newHashMapWithExpectedSize(overflowResults.size());
+            ClosableIterator<AgnosticLightResultRow> overflowIter = table.getOverflow(overflowResults.values());
+            try {
+                while (overflowIter.hasNext()) {
+                    AgnosticLightResultRow row = overflowIter.next();
+                    // QA-94468 LONG RAW typed columns ("val" in this case) must be retrieved first from the result set
+                    // see https://docs.oracle.com/cd/B19306_01/java.102/b14355/jstreams.htm#i1007581
+                    byte[] val = row.getBytes("val");
+                    long id = row.getLong("id");
+                    values.put(id, val);
+                }
+            } finally {
+                overflowIter.close();
+            }
+            for (Entry<Cell, OverflowValue> entry : overflowResults.entrySet()) {
+                Cell cell = entry.getKey();
+                OverflowValue ov = entry.getValue();
+                byte[] val = values.get(ov.id);
+                Preconditions.checkNotNull(val, "Failed to load overflow data: cell=%s, overflowId=%s", cell, ov.id);
+                results.put(cell, Value.create(val, ov.ts));
+            }
+        }
+        return results;
+    }
+
+    @Override
+    public Map<Cell, Long> getLatestTimestamps(TableReference tableRef, final Map<Cell, Long> timestampByCell) {
+        return runRead(tableRef, new Function<DbReadTable, Map<Cell, Long>>() {
+            @Override
+            @SuppressWarnings("deprecation")
+            public Map<Cell, Long> apply(DbReadTable table) {
+                ClosableIterator<AgnosticLightResultRow> iter = table.getLatestCells(timestampByCell, false);
+                try {
+                    Map<Cell, Long> results = Maps.newHashMap();
+                    while (iter.hasNext()) {
+                        AgnosticLightResultRow row = iter.next();
+                        Cell cell = Cell.create(row.getBytes("row_name"), row.getBytes("col_name"));
+                        long ts = row.getLong("ts");
+                        Long oldTs = results.put(cell, ts);
+                        if (oldTs != null && oldTs > ts) {
+                            results.put(cell, oldTs);
+                        }
+                    }
+                    return results;
+                } finally {
+                    iter.close();
+                }
+            }
+        });
+    }
+
+    private static final int INSERT_PARTITION_COUNT = 1000;
```

Comment:
would be better as configurable setting (feel free to do this later on another commit)